mountaineer/annotation_helpers.py

```diff
@@ -94,6 +94,11 @@ def resolve_types(current_type: type):
         # Always echo back the current type to make sure that everything that we've processed
         # is included in the final list
         yield resolve_forwardrefs(current_type, _globals=_globals, _locals=_locals)
+        # Python 3.10 resolving types behaved a little differently
+        origin = get_origin(current_type)
+        if origin:
+            current_type = get_origin(current_type)  # type: ignore[assignment]
```

Comment:
Can you explain more in your comment why 3.10 needs this (and why this is forward compatible with later python versions)? What tests weren't passing with this excluded?